package/harvester-os/files/usr/sbin/harv-install

```diff
@@ -180,29 +180,49 @@ preload_rke2_images()
       rke2_tmp="$inst_tmp/rke2"
       mkdir -p $rke2_tmp && tar xf $inst_tmp/rke2.linux-amd64.tar.gz -C $rke2_tmp
 
+      echo "Start rke2 and wait for containerd exist..."
       $rke2_tmp/bin/rke2 server &> /rke2.log &
 
+      until [ -f "/var/lib/rancher/rke2/bin/containerd" ] && [ -f "/var/lib/rancher/rke2/agent/etc/containerd/config.toml" ]
+      do
+        sleep 1
+      done
+
+      echo "Stop rke2 and remove tmporary rke2 files..."
+      pkill rke2
+      rm /rke2.log
+      rm /etc/rancher/rke2/rke2.yaml
+      rm -rf /var/lib/rancher/rke2/server
+      rm -rf /var/lib/rancher/rke2/agent/pod-manifests/*
+
       export PATH=/var/lib/rancher/rke2/bin:$PATH
       export CONTAINERD_ADDRESS=/run/k3s/containerd/containerd.sock
 
+      echo "Start containerd..."
+      containerd -c /var/lib/rancher/rke2/agent/etc/containerd/config.toml -a $CONTAINERD_ADDRESS --state /run/k3s/containerd --root /var/lib/rancher/rke2/agent/containerd &> /containerd.log &
```

Comment:
can u please share a little bit more about the RKE2 changes and why we need to start the container independently for now?

Reply:
Hi @guangbochen, this is what Kiefer encountered when he tried to boot four nodes simultaneously on the same disk. RKE2 server may go down before images are all loaded. We changed to start the RKE2 server to load contained binary and don't rely on RKE2 to load images to avoid the timeout issue.

Reply:
I tried to read RKE2 source code, but I can't find where may let the RKE2 server go down.

Reply:
maybe @brandond can u share some insights here, thanks?

Reply:
Previously, we bring up RKE2 server by calling `rke-server` and let RKE2 preload images in ZST tarballs. I'm not quite what changed, the rke2-server seems to timeout sooner if it can't see etcd ready.

Reply:
We had hoped that 8.5minutes would cover the time for the tarballs to be uncompressed. But I am happy to raise the timeout limit if you can give me a estimate of how much time.